jobs_archival/worker.go

```diff
@@ -0,0 +1,249 @@
+		toArchive = false
+	}
+	var statusList []*jobsdb.JobStatusT
+	if !toArchive {
+		statusList = getStatuses(
+			jobs,
+			func(*jobsdb.JobT) string { return jobsdb.Aborted.State },
+			[]byte(reason),
+		)
+		goto markStatus
+	}
+
+	location, err = w.uploadJobs(w.lifecycle.ctx, jobs)
+	if err != nil {
+		w.log.Errorf("failed to upload jobs - partition: %s - %w", w.partition, err)
+		maxAttempts := w.config.maxRetryAttempts()
+		statusList = getStatuses(
+			jobs,
+			func(job *jobsdb.JobT) string {
+				if job.LastJobStatus.AttemptNum >= maxAttempts {
+					return jobsdb.Aborted.State
+				}
+				return jobsdb.Failed.State
+			},
+			[]byte(fmt.Sprintf(`{"location": "not uploaded because - %v"}`, err)),
+		)
+		goto markStatus
+	}
+
+	statusList = getStatuses(
+		jobs,
+		func(*jobsdb.JobT) string { return jobsdb.Succeeded.State },
+		[]byte(fmt.Sprintf(`{"location": "%v"}`, location)),
+	)
+
+markStatus:
+	if err := w.jobsDB.UpdateJobStatus(w.lifecycle.ctx, statusList, nil, nil); err != nil {
+		w.log.Errorf("failed to mark jobs' status - %w", err)
+	}
+	if limitReached {
+		goto start
+	}
+	return true
+}
+
+func (w *worker) SleepDurations() (min, max time.Duration) {
+	return w.config.minSleep, time.Until(w.lastUploadTime.Add(w.config.maxSleep))
+}
+
+func (w *worker) Stop() {
+	w.lifecycle.cancel()
+}
+
+func (w *worker) uploadJobs(ctx context.Context, jobs []*jobsdb.JobT) (string, error) {
+	firstJobCreatedAt := jobs[0].CreatedAt
+	lastJobCreatedAt := jobs[len(jobs)-1].CreatedAt
+	workspaceID := jobs[0].WorkspaceId
+
+	w.log.Infof("[Archival: storeErrorsToObjectStorage]: Starting logging to object storage - %s", w.partition)
+
+	gzWriter := fileuploader.NewGzMultiFileWriter()
+	path := fmt.Sprintf(
+		"%v%v.json.gz",
+		lo.Must(misc.CreateTMPDIR())+"/rudder-backups/"+w.partition+"/",
+		fmt.Sprintf("%v_%v_%v", firstJobCreatedAt.Unix(), lastJobCreatedAt.Unix(), workspaceID),
+	)
+
+	for _, job := range jobs {
+		j, err := marshalJob(job)
+		if err != nil {
+			return "", fmt.Errorf("failed to marshal job - %w", err)
+		}
+		if _, err := gzWriter.Write(path, append(j, '\n')); err != nil {
+			return "", fmt.Errorf("failed to write to file - %w", err)
+		}
+	}
+	err := gzWriter.Close()
+	if err != nil {
+		return "", fmt.Errorf("failed to close file - %w", err)
+	}
+	defer func() { _ = os.Remove(path) }()
+
+	fileUploader, err := w.storageProvider.GetFileManager(workspaceID)
+	if err != nil {
+		w.log.Errorf("Skipping Storing errors for workspace: %s - partition: %s since no file manager is found",
+			workspaceID, w.partition,
+		)
+		return "", fmt.Errorf("failed to get file manager - %w", err)
+	}
+
+	file, err := os.Open(path)
+	if err != nil {
+		return "", fmt.Errorf("failed to open file %s - %w", path, err)
+	}
+	defer func() { _ = file.Close() }()
+	year, month, date := firstJobCreatedAt.Date()
+	prefixes := []string{
+		w.partition,
+		w.archiveFrom,
+		fmt.Sprintf("%d-%d-%d", year, month, date),
```

Comment:
Please use standard time formatting 
```suggestion
		firstJobCreatedAt.Format("2006-01-02"),
```